src/features/ResultsBlock/ResultsBlockView.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useState } from 'react';
+import { Link, useNavigate } from 'react-router-dom';
+import { Planet, ResultsBlockProps } from '../../types/interfaces';
+import Pagination from '../Pagination/Pagination';
+import DetailsPage from '../../pages/DetailPage/DetailPageView';
+
+const ResultsBlock: React.FC<ResultsBlockProps> = ({
+  searchResults,
+  countResults,
+  currentPage,
+  countPerPage,
+  setCurrentPage,
+}) => {
+  const navigate = useNavigate();
+  const [selectedPlanet, setSelectedPlanet] = useState<Planet | null>(null);
+  const handleBackClick = () => {
+    if (selectedPlanet) {
+      navigate(`/search/${currentPage}`);
+      setSelectedPlanet(null);
+    }
+  };
```

Comment:
looks like you need to define else case here

Reply:
needs to be discussed